components/widgets/menu/menu_items/submenu_item_action.jsx

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import * as Utils from 'utils/utils.jsx';
+import {ModalIdentifiers} from 'utils/constants';
+import SubMenuModal from '../../../submenu_modal/submenu_modal.jsx';
```

Comment:
Given that `SubMenuModal` component is directly related to the menu structure, I would suggest moving inside to `components/widgets/menu/menu_modals/submenu_modal`

Reply:
Sounds good.  Moved.